ui/main_window.cpp

```diff
@@ -568,7 +581,8 @@ bool MainWindow::InitializePlugins()
 void MainWindow::OnTraceAvailable(const QString &path)
 {
     qDebug() << "Trace is at " << path;
-    LoadFile(path.toStdString().c_str(), true);
+    // Figure out what do we do if we get repeated trigger of LoadFile before async call is done.
```

Comment:
Could you provide a little more context to this situation? OnTraceAvailable should only be triggered once a capture has been made. The user shouldn't be able to try and load a file in between closing the capture dialog and the file being loaded to the UI

Reply:
Is there anything preventing user from capturing multiple traces in rapid succession?